ycsb_wrapper/trigger_ycsb.py

```diff
@@ -0,0 +1,149 @@
+                    summary_dict[summ[0].strip('[').strip(']')] = {}
+                summary_dict[summ[0].strip('[').strip(']')][summ[1]] = float(summ[2])
+            summary.append({
+                "workload": "ycsb",
+                "uuid": uuid,
+                "user": user,
+                "phase": phase,
+                "driver": driver,
+                "timestamp": datetime.now(),
+                "data": summary_dict,
+                "recordcount": int(recordcount),
+                "operationcount": int(operationcount),
+                "iteration": int(iteration),
+                "workload_type": workload
+            })
+        return processed, summary
+
+    def _parse_stdout(self, data):
+        data_points = re.findall(
+            r"(\d+-\d+-\d+ \d+:\d+:\d+:\d+) \d+ sec: (\d+ operations); (\d+.\d+? current ops/sec); (.*)",
+            data)
+        summary = re.findall(r"(.*), (.*), (.*)", data)
+        return {"results": data_points, "summary": summary}
+
+    def emit_actions(self):
+        extra = ""
+        if self.extra is not None:
+            extra = self.extra[0]
+        python = "/usr/bin/python2"
+        if self.load:
+            phase = "load"
+            cmd = "{} /ycsb/bin/ycsb {} {} -s -P /tmp/ycsb/{} {}".format(python, phase,
+                                                                         self.driver[0],
+                                                                         self.workload[0],
+                                                                         extra)
+            stdout, stderr, rc = self._run(cmd)
+            output = "{}\n{}".format(stdout, stderr)
+        else:
+            phase = "run"
+            cmd = "{} /ycsb/bin/ycsb {} {} -s -P /tmp/ycsb/{} {}".format(python, phase,
+                                                                         self.driver[0],
+                                                                         self.workload[0],
+                                                                         extra)
+            stdout, stderr, rc = self._run(cmd)
+            output = "{}\n{}".format(stdout, stderr)
+
+        if rc != 0:
+            print("YCSB failed to execute:\n%s", stderr)
+            exit(1)
+        if "Error inserting" in stderr:
+            print("YCSB failed to load database... Drop previous YCSB database")
+            exit(1)
+
+        data = self._parse_stdout(output)
+        print(output)
+        documents, summary = self._json_payload(data, self.run[0], self.uuid, self.user,
+                                                self.phase, self.workload,
+                                                self.driver[0],
+                                                self.recordcount, self.operationcount,
+                                                self.cluster_name)
+
+        print("Attempting to index results...")
+        if len(documents) > 0:
```

Comment:
There's a few errors here, it should be:
```
this needs to be:
```
            for document in documents:
                yield document, 'results'
            for document in summary:
                 yield document, 'summary'
```
We've to yield the individual document and not the list.